src/objects/core/zcl_abapgit_item_graph.clas.abap

```diff
@@ -49,25 +57,28 @@ CLASS ZCL_ABAPGIT_ITEM_GRAPH IMPLEMENTATION.
 
 
   METHOD constructor.
-    INSERT LINES OF it_items INTO TABLE mt_vertices.
+    mt_vertices = it_items.
   ENDMETHOD.
 
 
   METHOD get_next.
 * find a vertex with no inbound edges, if it does not exist pick anything
 
-    DATA ls_vertex LIKE LINE OF mt_vertices.
     DATA lv_index  TYPE i.
+    FIELD-SYMBOLS: <ls_vertex> TYPE ty_vertex.
 
-    LOOP AT mt_vertices INTO ls_vertex.
+    LOOP AT mt_vertices ASSIGNING <ls_vertex>
+                        WHERE has_inbound_edge = abap_false.
       lv_index = sy-tabix.
-      READ TABLE mt_edges WITH KEY
-        to-obj_type = ls_vertex-obj_type
-        to-obj_name = ls_vertex-obj_name
+      READ TABLE mt_edges WITH KEY sec_key COMPONENTS
+        to-obj_type = <ls_vertex>-obj_type
+        to-obj_name = <ls_vertex>-obj_name
         TRANSPORTING NO FIELDS.
-      IF sy-subrc <> 0.
+      IF sy-subrc = 0.
+        <ls_vertex>-has_inbound_edge = abap_true.
```

Comment:
`has_inbound_edge` is not reset anywhere?

Reply:
yeah, doesn't work that way.

Reply:
a counter for the number of edges might work

Reply:
Didn't understand the code well enough to make that fix. Will come up with another solution. Maybe secondary keys are enough.